qiskit/quantum_info/states/stabilizerstate.py

```diff
@@ -0,0 +1,591 @@
+        """
+
+        num_qubits = self.data.num_qubits
+
+        # Check if there exists stabilizer anticommuting with Z[qubit]
+        # in this case the measurement outcome is random
+        z_anticommuting = np.nonzero(self.data.stabilizer.X[:, qubit])[0]
+
+        # Non-deterministic outcome
+        if len(z_anticommuting) != 0:
+            p_qubit = np.min(np.nonzero(self.data.stabilizer.X[:, qubit]))
+            p_qubit += num_qubits
+            outcome = randbit
+
+            # Updating the StabilizerState
+            for i in range(2 * num_qubits):
+                # the last condition is not in the AG paper but we seem to need it
+                if (self.data.table.X[i][qubit]) and (i != p_qubit) and \
+                        (i != (p_qubit - num_qubits)):
+                    self._rowsum_nondeterministic(i, p_qubit)
+
+            self.data.table[p_qubit - num_qubits] = copy.deepcopy(self.data.table[p_qubit])
+            self.data.table.X[p_qubit] = np.zeros(num_qubits)
+            self.data.table.Z[p_qubit] = np.zeros(num_qubits)
+            self.data.table.Z[p_qubit][qubit] = True
+            self.data.table.phase[p_qubit] = outcome
+            return outcome
+
+        # Deterministic outcome - measuring it will not change the StabilizerState
+        aux_pauli = Pauli(num_qubits * 'I')
+        for i in range(num_qubits):
+            if self.data.table.X[i][qubit]:
+                aux_pauli = self._rowsum_deterministic(aux_pauli, i + num_qubits)
+        outcome = aux_pauli.phase
+        return outcome
+
+    @staticmethod
+    def _phase_exponent(x1, z1, x2, z2):
+        """Exponent g of i such that Pauli(x1,z1) * Pauli(x2,z2) = i^g Pauli(x1+x2,z1+z2) """
+        # pylint: disable=invalid-name
+
+        phase = (x2 * z1 * (1 + 2 * z2 + 2 * x1) - x1 * z2 * (1 + 2 * z1 + 2 * x2)) % 4
+        if phase < 0:
+            phase += 4  # now phase in {0, 1, 3}
+
+        if phase == 2:
+            raise QiskitError(
+                'Invalid rowsum phase exponent in measurement calculation.')
+        return phase
+
+    def _rowsum(self, accum_pauli, accum_phase, row_pauli, row_phase):
+        """Aaronson-Gottesman rowsum helper function """
+
+        newr = 2 * row_phase + 2 * accum_phase
+
+        for qubit in range(self.data.num_qubits):
+            newr += self._phase_exponent(row_pauli.x[qubit],
+                                         row_pauli.z[qubit],
+                                         accum_pauli.x[qubit],
+                                         accum_pauli.z[qubit])
+        newr %= 4
+        if (newr != 0) & (newr != 2):
+            raise QiskitError(
+                'Invalid rowsum in measurement calculation.')
+
+        accum_phase = int((newr == 2))
+        accum_pauli.x ^= row_pauli.x
+        accum_pauli.z ^= row_pauli.z
+        return accum_pauli, accum_phase
+
+    def _rowsum_nondeterministic(self, accum, row):
```

Comment:
This should also be a `@staticmethod` and replace `self` with the `table` to update:
```suggestion
    @staticmethod
    def _rowsum_nondeterministic(table, accum, row):
```